### PR TITLE
Update installation instructions on Apple Silicon, suppress linker warnings

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

With these changes I can compile spectre on an M2 Mac.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
Moved a commit to #4708

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsvu** [comment]
@geoffrey4444 could you review this please, and check that the linker warnings are gone when you build on your M1 machine?

File: docs/Installation/InstallationOnAppleSilicon.md [hunk 1]
```diff
@@ -81,14 +71,17 @@ following the instructions on the [homebrew](https://brew.sh) homepage. Then,
...
+brew install boost gsl cmake doxygen
 brew install ccache autoconf automake jemalloc hdf5 pybind11 yaml-cpp
 ```
 
 ### 4. Install remaining dependencies
 
 Here, install the remaining dependencies that cannot be installed
-with homebrew or miniforge.
+with homebrew or miniforge. You can install them from source manually, or use
+the [Spack](https://github.com/spack/spack) package manager.
```
Comment (geoffrey4444) [inline docs/Installation/InstallationOnAppleSilicon.md:82]: "You can also install them from source manually..."

**Comment by nilsvu** [comment]
@geoffrey4444 As discussed I just suppress linker warnings on macOS now, and added some explanations in the CMake file. Please take a look when you get the chance!

**Review by nilsdeppe — APPROVED** [review]

**Review by geoffrey4444 — APPROVED** [review]

File: docs/Installation/InstallationOnAppleSilicon.md (hunk 1 above)
Comment (nilsvu) [reply docs/Installation/InstallationOnAppleSilicon.md:82]: This doesn't quite fit in the sentence, or maybe I'm misunderstanding it?

Comment (nilsvu) [reply docs/Installation/InstallationOnAppleSilicon.md:82]: merging now, let me know if you want me to reword anything in a follow-up

**Pull request merged**